### PR TITLE
Put branch cache table into table group for consistent logic handling

### PR DESCRIPTION
**Pull request by shifucun**
Current server saves an explicit branch table object and the bigtable.Read() function returns data for branch table explicitly. As now we have merging for all the cache with import group changes, branch cache table is no special at all. 

This PR put the branch cache table (even in non import-group mode) into the table group and let the general table merge logic to handle it. In fact, in import group mode, the branch cache is jus a regular group in all the tables.

### CODE REVIEW

File: test/e2e/setup.go [hunk 1]
```diff
@@ -105,29 +106,29 @@ func setupInternal(
...
+		if err != nil {
+			return nil, nil, err
+		}
+		tables = append(tables, bigtable.NewTable(name, baseTable))
+		branchTable, err := createBranchTable(ctx)
 		if err != nil {
 			return nil, nil, err
 		}
-		baseTables = append(baseTables, baseTable)
+		tables = append(tables, bigtable.NewTable(branchTableName, branchTable))
```
Comment (pradh) [inline test/e2e/setup.go:131]: Could you maybe add a few lines to the description that clarify why this change is safe in current prod?

File: internal/store/bigtable/group.go [hunk 2]
```diff
@@ -34,64 +34,104 @@ var groupRank = map[string]int{
...
+	tables := []*Table{}
+	for _, t := range g.tables {
+		if t.name != g.branchTableName {
+			tables = append(tables, t)
+		}
+	}
+	tables = append(tables, branchTable)
+	g.branchTableName = branchTable.name
+	g.tables = tables
+	if g.isProto {
```
Comment (pradh) [inline internal/store/bigtable/group.go:116]: else, is the assertion that branch cache always come after base?  Wonder if it's worth asserting?

**Review by pradh — APPROVED** [review]

File: internal/store/bigtable/group.go (hunk 2 above)
Comment (shifucun) [reply internal/store/bigtable/group.go:117]: Done

File: test/e2e/setup.go (hunk 1 above)
Comment (shifucun) [reply test/e2e/setup.go:131]: Done
